devicetypes/smartthings/orvibo-Moisture-Sensor.src/orvibo-Moisture-Sensor.groovy

```diff
@@ -0,0 +1,174 @@
+ */
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+import physicalgraph.zigbee.zcl.DataType
+
+metadata {
+	definition(name: "Orvibo Moisture Sensor", namespace: "smartthings", author: "SmartThings", vid: "generic-leak", mnmn:"SmartThings", ocfDeviceType: "x.com.st.d.sensor.moisture") {
+		capability "Configuration"
+		capability "Refresh"
+		capability "Water Sensor"
+		capability "Sensor"
+		capability "Health Check"
+		capability "Alarm"
+
+		fingerprint profileId: "0104",deviceId: "0402",inClusters: "0000,0003,0500,0001,0009", outClusters: "0019", manufacturer: "Heiman", model: "2f077707a13f4120846e0775df7e2efe"
+	}
+
+	simulator {
+
+		status "dry": "zone status 0x0020 -- extended status 0x00"
+		status "wet": "zone status 0x0021 -- extended status 0x00"
+
+		for (int i = 0; i <= 90; i += 10) {
+			status "battery 0021 0x${i}": "read attr - raw: 8C900100010A21000020C8, dni: 8C90, endpoint: 01, cluster: 0001, size: 0A, attrId: 0021, result: success, encoding: 20, value: ${i}"
+		}
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"water", type: "generic", width: 6, height: 4){
+			tileAttribute ("device.water", key: "PRIMARY_CONTROL") {
+				attributeState "dry", icon:"st.alarm.water.dry", backgroundColor:"#ffffff"
+				attributeState "wet", icon:"st.alarm.water.wet", backgroundColor:"#00a0dc"
+			}
+		}
+
+		standardTile("alarm", "device.alarm", width: 2, height: 2) {
+			state "off", label:'off', action:'alarm.both', icon:"st.alarm.alarm.alarm", backgroundColor:"#ffffff"
+			state "strobe", label:'strobe!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "siren", label:'siren!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+			state "both", label:'alarm!', action:'alarm.off', icon:"st.alarm.alarm.alarm", backgroundColor:"#e86d13"
+		}
+
+		valueTile("battery", "device.battery", decoration: "flat", inactiveLabel: false, width: 2, height: 2) {
+			state "battery", label:'${currentValue}% battery', unit:""
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main "water"
+		details(["water", "alarm", "battery", "refresh"])
+	}
+}
+
+def parse(String description) {
+	log.debug "description: $description"
+
+	def result = [:]
+	Map map = zigbee.getEvent(description)
+
+	if (!map) {
+		if (description?.startsWith('zone status')) {
+			map = getMoistureResult(description)
+			result = createEvent(map)
+		} else if(description?.startsWith('enroll request')){
+			List cmds = zigbee.enrollResponse()
+			log.debug "enroll response: ${cmds}"
+			result = cmds?.collect { new physicalgraph.device.HubAction(it) }
+		}else {
+			Map descMap = zigbee.parseDescriptionAsMap(description)
+			if (descMap?.clusterInt == 0x0500 && descMap.attrInt == 0x0002) {
+				map = getMoistureResult(description)
+				result = createEvent(map)
```

Comment:
is there any way you can refactor this code not to have 4 `createEvent` calls in one method?